packages/client/src/__tests__/integration/happy/exhaustive-schema-mongo/__snapshots__/binary.test.ts.snap

```diff
@@ -1000,25 +1000,25 @@ export namespace Prisma {
    */
   export import Decimal = runtime.Decimal
 
-  export type DecimalJsLike = runtime.DecimalJsLike
+  export import DecimalJsLike = runtime.DecimalJsLike
 
   /**
    * Metrics 
    */
-  export type Metrics = runtime.Metrics
-  export type Metric<T> = runtime.Metric<T>
-  export type MetricHistogram = runtime.MetricHistogram
-  export type MetricHistogramBucket = runtime.MetricHistogramBucket
+  export import Metrics = runtime.Metrics
+  export import Metric = runtime.Metric
+  export import MetricHistogram = runtime.MetricHistogram
+  export import MetricHistogramBucket = runtime.MetricHistogramBucket
 
   /**
   * Extensions
   */
-  export type Extension = $Extensions.UserArgs
+  export import Extension = $Extensions.UserArgs
   export import getExtensionContext = runtime.Extensions.getExtensionContext
-  export type Args<T, F extends $Public.Operation> = $Public.Args<T, F>
-  export type Payload<T, F extends $Public.Operation> = $Public.Payload<T, F>
-  export type Result<T, A, F extends $Public.Operation> = $Public.Result<T, A, F>
-  export type Exact<T, W> = $Public.Exact<T, W>
+  export import Args = $Public.Args
+  export import Payload = $Public.Payload
+  export import Result = $Public.Result
+  export import Exact = $Public.Exact
```

Comment:
Important part of the fix. I prevents the expansion of our types into the user's projects - does not work for recursive types. With this, a compiled user project using an extension would output types like

```ts
    $allModels: {
        paginate<T, A>(this: T, args: Prisma.Exact<A, Prisma.Args<T, "findMany">>): Promise<Prisma.Result<T, A, "findMany">>;
    };
```

With the previous version, tsc will actually try to inline the types with they actual implementation and potentially resolve them in-place depending on their input. This would be great if it did not actually produce a cryptic error message and could inline properly. tl;dr; only re-export via `export import A = B`.

> The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.

Reply:
Actually: Apparently this is not valid in TypeScript, although it produces the desired outcome once compiled.

Reply:
So I've used a workaround that we used for this same issue in the past. Intersect the type with ` & unknown` prevents TS from performing type inlining in dependent projects. Ideally, we would want something like this https://github.com/microsoft/TypeScript/issues/4336, but the current "[solution](https://github.com/microsoft/TypeScript/issues/4336#issuecomment-781862959)" we also use does not work with types, which was an assumption in my first comment.

Reply:
Related slack thread https://prisma-company.slack.com/archives/C043X6JBND6/p1689103849934459

Reply:
Decided to use the first approach, and suppress the error message (which only happen in watch mode locally). The error is never suppressed on CI.

Reply:
Can it conflict with user defined models now? I.e. is it still possible to name your model `Result`? Could you add those names [here](https://github.com/prisma/prisma/blob/main/packages/client/tests/functional/naming-conflict/_builtInNames.ts)  to verify?

Reply:
I guess before and after will both conflict, I'll write a test.

Reply:
Turns out no there's no conflict. Did I understand correctly that there was no conflict because `Result` is in `Prisma` namespace and the model type definition was top-level?

Reply:
Tests added.

Reply:
>Turns out no there's no conflict. Did I understand correctly that there was no conflict because Result is in Prisma namespace and the model type definition was top-level?

This, and also within `Prisma` namespace we never refer to model type by the name anymore.